kokoro/build_binaries_rust.sh

```diff
@@ -42,8 +42,8 @@ readonly generated_binaries=(
     ./enclave_apps/target/x86_64-unknown-none/release/oak_orchestrator
 )
 readonly binary_names=(
-    oak_restricted_kernel_simple_io_wrapper_bin
-    oak_restricted_kernel_simple_io_init_rd_wrapper_bin
+    oak_restricted_kernel_simple_io_bzimage
```

Comment:
I personally think we should probably rename the internal binary names to be clearer. But let's not do this right now.

Updated this PR to match established internal binary names.

Reply:
> But I don't understand how this resolves the problem discussed - in artifacts_path we have the * (wildcard) which matches three files. One of the files (name needs to be configurable) is special, since it is the one for the internal TR binary package. I don't see a fundamentally different way than https://github.com/project-oak/oak/pull/5001 to fix this.

With this PR:
- the name of this binary remains the same as it used to be prior to https://github.com/project-oak/oak/pull/4976 etc.
- the tag under which its provenance is uploaded to ent remains the same.
- it remains listed as a subject in this provenance under the same name. 


With the additional files (image and setup data) there's now some additional tags under which the provenance will be uploaded. But TR can ignore those. 

There's now multiple subjects in the provenance, but TR can find the right one by name. 

I might be missing something here, but I don't think there's a fundamentally breaking change that means we need to add new data to the config. We used filenames as identifiers in the past, and we still do so.

Reply:
OK I get it now. The bzImage file name is constructed such that it coincides with the package name. Good. So assume we can remove `binary_file` everywhere in #5001. But how about buildconfigs/oak_containers_kernel_bzimage.toml? This also needs a split, in exactly the same way  The package name is oak_containers_kernel.

To get the package_name into the reusable_provenance.yaml we *could* also use the basename of the buildconfig (need to rename buildconfigs appropriately). Or we add it to the matrix in provenance.yaml. I can find a better way here.

Reply:
> The bzImage file name is constructed such that it coincides with the package name. Good. So assume we can remove binary_file everywhere in https://github.com/project-oak/oak/pull/5001.

yup!

> But how about buildconfigs/oak_containers_kernel_bzimage.toml? This also needs a split, in exactly the same way The package name is oak_containers_kernel.

Yes, eventually we'll want to extend it's provenance to include the image and setup data also. I'm not sure I understand why we'd need the changes in #5001 should be needed for that, but I'd be happy to sync more on Monday. :) 

Right now there haven't yet been any changes to the containers kernel.